File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/CBPacketTime.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.net.tcp.packet.PacketHandler;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;

import java.io.IOException;

public class CBPacketTime extends IPacket<PacketHandler>{
    
    public static final int PACKET_ID = 22;
    
    public CBPacketTime(){
        super(PACKET_ID);
    }
    
    
    public long gameTimeTicks;
    
    public CBPacketTime(long gameTimeTicks){
        this();
        this.gameTimeTicks = gameTimeTicks;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeLong(gameTimeTicks);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        gameTimeTicks = stream.readLong();
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/CBPacketTime.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.net.tcp.packet.PacketHandler
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class CBPacketTime() : IPacket<PacketHandler?>(PACKET_ID) {
    var gameTimeTicks: Long = 0

    constructor(gameTimeTicks: Long) : this() {
        this.gameTimeTicks = gameTimeTicks
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeLong(gameTimeTicks)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        gameTimeTicks = stream!!.readLong()
    }

    companion object {
        const val PACKET_ID = 22
    }
}